src/main/java/org/folio/innreach/domain/service/impl/CirculationServiceImpl.java

```diff
@@ -262,6 +268,37 @@ public InnReachResponseDTO returnUncirculated(String trackingId, String centralC
     }
   }
 
+  @Override
+  public InnReachResponseDTO recall(String trackingId, String centralCode, RecallDTO recallDTO) {
+    var transaction = getTransaction(trackingId, centralCode);
+    var requestId = transaction.getHold().getFolioRequestId();
+    var request = requestService.findRequest(requestId);
+
+    try {
+      if (request.getStatus() == RequestDTO.RequestStatus.OPEN_AWAITING_PICKUP ||
+        request.getStatus() == RequestDTO.RequestStatus.OPEN_IN_TRANSIT) {
+        requestService.cancelRequest(transaction, "Item has been recalled.");
+      } else {
+        var recallUser = getRecallUserForCentralServer(centralCode);
+        requestService.createRecallRequest(recallUser.getUserId(), transaction.getHold().getFolioItemId());
+      }
+    } catch (Exception e) {
+      throw new IllegalArgumentException(e.getMessage());
+    }
```

Comment:
Since these are actually two different scenarios, lets handle exceptions of them in two different try/catch blocks with an appropriate message for each. `Unable to create a recall request on item ... : + e.getMessage()` and another one for cancellation logic